src/core/layout/qgslayoutexporter.cpp

```diff
@@ -1275,6 +1281,50 @@ void QgsLayoutExporter::preparePrintAsPdf( QgsLayout *layout, QPagedPaintDevice
 
   updatePrinterPageSize( layout, device, firstPageToBeExported( layout ) );
 
+  // force a non empty title to avoid invalid (according to specification) PDF/X-4
+  const QString title = layout->project()->metadata().title().isEmpty() ?
```

Comment:
There's a project nullptr check later in this method, so better check it here too:

```suggestion
  const QString title = !layout->project() || layout->project()->metadata().title().isEmpty() ?
```